### PR TITLE
Fix non-determinism in TLMonitor generation

### PR DESCRIPTION
**Pull request by jackkoenig**
groupBy considered harmful

**Related issue**: 

**Type of change**: bug fix

**Impact**: no functional change / large impact on generated code order

**Development Phase**: implementation

**Release Notes**

Fix non-determinism in TLMonitor generation

### CODE REVIEW

**Comment by jackkoenig** [comment]
*note* This has a large impact on the order, perhaps worth playing with to have smaller diff

File: src/main/scala/tilelink/Parameters.scala [hunk 1]
```diff
@@ -161,9 +161,11 @@ case class TLManagerPortParameters(
       lgSize:  UInt,
       range:   Option[TransferSizes]): Bool = {
     def trim(x: TransferSizes) = range.map(_.intersect(x)).getOrElse(x)
+    // groupBy returns an unordered map, convert back to Seq and sort the result for determinism
     val supportCases = managers.groupBy(m => trim(member(m))).mapValues(_.flatMap(_.address))
```
Comment (terpstra) [inline src/main/scala/tilelink/Parameters.scala]: Why not just ask it to return a SortedMap instead?

Comment (jackkoenig) [reply src/main/scala/tilelink/Parameters.scala]: There's no implicit ordering for TransferSizes so I thought it easier just to go to Seq and then use `.sortWith`. Further more, supportCases is not actually used as a Map so I thought it more clear to make it a Seq

Comment (terpstra) [reply src/main/scala/tilelink/Parameters.scala]: Ok, fine.

**Pull request merged**